[v7] Bump `@noble` libs to v2

larabr opened the pull request:
See commits for more details.
The new lib versions include support for md5 and brainpool curves, so we can drop our internal implementations.

NB: the libs require bumping the min supported version from Node 18 to Node 20, hence this change will be released in  OpenPGP.js v7 .

Supersedes #1911, #1912 , #1913 .

<details><summary>Bundled lib size changes</summary>
Before:
<img width="1510" height="778" alt="image" src="https://github.com/user-attachments/assets/f9c016ec-8649-47cf-bc32-b05b7357ddff" />

After (standalone brainpool modules were ~2KB) : 
<img width="1510" height="778" alt="image" src="https://github.com/user-attachments/assets/6dc2eeab-e921-4c8f-88ea-6d32e950d0a6" />

</details>

dependabot[bot] committed:
Bump @noble/curves to v2

Co-authored-by: Daniel Huigens <d.huigens@protonmail.com>

dependabot[bot] committed:
Bump @noble/hashes to v2

dependabot[bot] committed:
Bump @noble/ciphers to v2

larabr commented on src/crypto/public_key/elliptic/ecdsa.js:
```diff
@@ -26,7 +26,6 @@ import util from '../../../util.js';
 import { getRandomBytes } from '../../random.js';
```
Should double check this padding issue: https://github.com/openpgpjs/openpgpjs/actions/runs/29019822522/job/86123892418?pr=1919

twiss replied:
WebKit on Windows seems to sometimes generate P-521 private keys that are one byte short. We could pad them manually but it seems safer to just add a check and fallback to JS: https://github.com/openpgpjs/openpgpjs/pull/1919/commits/79ae63bc871b407185766dfa4ad485794b6edc49

larabr replied:
The error was thrown here actually I think: https://github.com/openpgpjs/openpgpjs/blob/2ac0048404b74a3595d503125b53f3b3d0486bec/src/crypto/public_key/elliptic/ecdh.js#L220 .

So let's just pad the key there; we already do the same on key [generation](https://github.com/openpgpjs/openpgpjs/blob/2ac0048404b74a3595d503125b53f3b3d0486bec/src/crypto/public_key/elliptic/oid_curves.js#L211) and parsing; this one-byte-off thing is not uncommon for p521. 

(I think on key generation we could/should limit it to pad at most one byte though, and only for this specific curve)

twiss replied:
> The error was thrown here actually I think:

Yes, and that in turn calls out to `webGenKeyPair` here: https://github.com/openpgpjs/openpgpjs/blob/2ac0048404b74a3595d503125b53f3b3d0486bec/src/crypto/public_key/elliptic/oid_curves.js#L177 (even though we're in `jsPublicEphemeralKey`, Web Crypto is still used for keygen if available).

> So let's just pad the key there; we already do the same on key [generation](https://github.com/openpgpjs/openpgpjs/blob/2ac0048404b74a3595d503125b53f3b3d0486bec/src/crypto/public_key/elliptic/oid_curves.js#L211) and parsing; this one-byte-off thing is not uncommon for p521.

I think padding here is potentially dangerous. What if the implementation accidentally returns a P-256 sized key for P-521, should we ignore it too?

The format for JWK strictly requires a fixed length, here: https://www.rfc-editor.org/info/rfc7518/#section-6.2.2.1. So, the Web Crypto implementation of WebKit on Windows is just wrong and should be fixed, instead; I'll file a bug with them.

larabr replied:
Windows webkit is not the only one that returns 1-byte short keys, I've seen the failure in Node and Safari before. If the curve is P-521 it's fine to pad the last byte, the noble lib is just being stricter than before about this it seems.

For other curves, or if the padding required is larger than 1 byte, we should just let the noble lib throw and not silently fallback to JS -- we've only been keeping fallbacks for WebCrypto for specific documented bugs.

larabr committed:
CurveWithOID: ecdh: pad ephemeral secret to expected payloadSize

Since the new noble lib no longer allows P-521 secrets of 65 bytes:
https://github.com/paulmillr/noble-curves/commit/b9d6fa30adff95731f79e9fb171e58fe12d429d3#diff-a7bb904f12070014203a286cf2b706212b313aa1dd3fc1fbaa61b26fedb59772L245
